### PR TITLE
Fix: Windows Build Instructions

### PR DESCRIPTION
**Pull request by Spidy123222**
This Pr fixes the build instructions for Xemu on windows to build. At first, found out that when trying to use the build command it wouldn't execute unless the backslashes were removed so it was removed in this case and seems to scale fine on the site. It was not getting the right type of headers to be able to build within the docker that Xemu uses for building and by changing it to the one Linux uses made it build. noticing that having a problem getting the docker to start getting recurse-submodules and was able to build successfully.

Site:
Screenshot 2022-02-03 161306

### CODE REVIEW

**Comment by nalabrie** [comment]
I built xemu on Windows 10 using the instructions that are currently on the website yesterday and it worked fine for me. Were you running the commands from within WSL2 Ubuntu or in a Windows cmd prompt / powershell? I'm pretty sure `\` is always understood by the bash shell at the end of the line to allow continuing the command on the next line. Which makes me wonder if the command was accidentally typed in cmd/powershell which probably doesn't understand that syntax (I'm unsure).

I have no comment on if `--recurse-submodules` is necessary, perhaps I was lucky and already had the right submodules installed on my machine. I suppose it couldn't hurt to run `--recurse-submodules` anyway to be safe, right?

**Comment by Spidy123222** [comment]
> I built xemu on Windows 10 using the instructions that are currently on the website yesterday and it worked fine for me. Were you running the commands from within WSL2 Ubuntu or in a Windows cmd prompt / powershell? I'm pretty sure `\` is always understood by the bash shell at the end of the line to allow continuing the command on the next line. Which makes me wonder if the command was accidentally typed in cmd/powershell which probably doesn't understand that syntax (I'm unsure).
> 
> 
> 
> I have no comment on if `--recurse-submodules` is necessary, perhaps I was lucky and already had the right submodules installed on my machine. I suppose it couldn't hurt to run `--recurse-submodules` anyway to be safe, right?

Honestly had no idea that was supposed to be ran on wsl2 terminal (not really mentioned to use it) and was doing this via powershell (likely why it was sharing folder to docker making it super slow). Could modify the instructions to use it in wsl2 terminal. This method does it in powershell/command prompt. 

From what I saw on the Linux build it seemed like it doesn't utilize the back slashes but are in similar lengths. It's possible that --recursive-submodules likely is needed via windows but it might be safe to do it that way since it seems xemu might be using submodules anyways i am no expert on that though.

**Comment by Spidy123222** [comment]
so I am attempting using the Ubuntu wsl2 terminal and it seems to build perfectly and it does seem to catch submodules if using `--recurse-submodules`

Screenshot 2022-02-04 183121

File: docs/docs/building-from-source.md [hunk 1]
```diff
@@ -5,15 +5,15 @@ Users are recommended to use the [pre-built xemu binaries](https://github.com/mb
 ## Windows
 
 Windows builds are cross-compiled from Ubuntu. If you would like to build *on* Windows, you can use WSL2 and Docker. See [official Docker
-documentation](https://docs.docker.com/docker-for-windows/wsl/) for how to get WSL2 and Docker set up.
+documentation](https://docs.docker.com/docker-for-windows/wsl/) for how to get WSL2 and Docker set up. In the instructions below for windows, It can be used in Powershell/Command prompt but is recommended to do this in your wsl2 terminal on windows.
```
Comment (MasonT8198) [inline docs/docs/building-from-source.md:8]: NIT: "Windows" not "windows" and "WSL2" not "wsl2"

File: docs/docs/building-from-source.md [hunk 2]
```diff
@@ -5,15 +5,15 @@ Users are recommended to use the [pre-built xemu binaries](https://github.com/mb
...
 
 ```bash
+# Configure git NOTE: Change back To true after use if needed
```
Comment (MasonT8198) [inline docs/docs/building-from-source.md:11]: Why is this necessary? I haven't had to do this on my Windows Docker setup, I don't think this needs to be added

File: docs/docs/building-from-source.md [hunk 3]
```diff
@@ -5,15 +5,15 @@ Users are recommended to use the [pre-built xemu binaries](https://github.com/mb
...
+git config --global core.autocrlf false
+
 # Clone and build
-git clone https://github.com/mborgerson/xemu.git
-docker run --rm -v $PWD/xemu:/xemu -w /xemu \
-    -e CCACHE_DIR=/xemu/ccache \
-    mborgerson/xemu-ubuntu-win64-cross:latest \
-    ./build.sh -p win64-cross
+git clone --recurse-submodules https://github.com/mborgerson/xemu.git
```
Comment (MasonT8198) [inline docs/docs/building-from-source.md:15]: Nit: --recurse-submodules shouldn't be needed here

**Review by MasonT8198 — CHANGES_REQUESTED** [review]

File: docs/docs/building-from-source.md (hunk 2 above)
Comment (Spidy123222) [reply docs/docs/building-from-source.md:11]: When running under powershell without that configuration change. It doesn't build at all.

File: docs/docs/building-from-source.md (hunk 3 above)
Comment (Spidy123222) [reply docs/docs/building-from-source.md:15]: On powershell when doing --recurse-submodules for cloning. It allowed it to build to the end.

File: docs/docs/building-from-source.md (hunk 2 above)
Comment (MasonT8198) [reply docs/docs/building-from-source.md:11]: testing on windows, without this it builds perfectly fine, are you sure that this isn't just a misnomer

File: docs/docs/building-from-source.md (hunk 3 above)
Comment (MasonT8198) [reply docs/docs/building-from-source.md:15]: This shouldn't affect anything, recursive sub modules doesn't pull anything extra

File: docs/docs/building-from-source.md (hunk 2 above)
Comment (Spidy123222) [reply docs/docs/building-from-source.md:11]: Yes Infact I couldn't build without it at all. It would just not get the docker image for building and if the image is there already in docker and working I haven't tested that part because getting it to get the image in first place wasn't working via powershell. I'm very positive it wasn't a misnomer. The only method I found that works without it is using the wsl2 terminal. When doing these commands in powershell with git for windows it fails building. Turning it off however wouldn't effect Linux since it already is using line endings it properly needs.

Comment (Spidy123222) [reply docs/docs/building-from-source.md:11]: This is currently what happens when both autocrlf and recurse-submodules is removed.

Current (autocrlf on and no submodules):
Screenshot 2022-02-05 171850
d

with autocrlf off no submodules, it doesn't finish building and just ends close to the end. (don't mind the build time used an old pc that was slow. It does the same thing on every Windows pc I try.): 
Screenshot 2022-02-05 202413

having recurse-submodules lets it finish to end with a successful exe.

**Comment by mborgerson** [comment]
Can you change this s.t. we just recommend people use the WSL2 console with bash. I really don't want to touch powershell.

**Comment by Spidy123222** [comment]
> Can you change this s.t. we just recommend people use the WSL2 console with bash. I really don't want to touch powershell.

Is this more of an adequate fix?

I added the sentence to use wsl2 terminal. Since someone  could just think they would be using powershell (like how I did) to do these instructions to their docker via powershell which partially works. 

Should be more clear now.

File: docs/docs/building-from-source.md [hunk 4]
```diff
@@ -5,15 +5,12 @@ Users are recommended to use the [pre-built xemu binaries](https://github.com/mb
 ## Windows
 
 Windows builds are cross-compiled from Ubuntu. If you would like to build *on* Windows, you can use WSL2 and Docker. See [official Docker
-documentation](https://docs.docker.com/docker-for-windows/wsl/) for how to get WSL2 and Docker set up.
+documentation](https://docs.docker.com/docker-for-windows/wsl/) for how to get WSL2 and Docker set up. The instructions are below for windows. Make sure to be using the wsl2 terminal for the instructions.
```
Comment (mborgerson) [inline docs/docs/building-from-source.md:8]: Don't need the line `The instructions are below for windows.`

Comment (Spidy123222) [reply docs/docs/building-from-source.md]: Resolved on latest commit, Is that better?
